engine/src/button.cpp

```diff
@@ -2735,6 +2735,11 @@ void MCButton::openmenu(Boolean grab)
 				/* UNCHECKED */ MCStringFirstIndexOfChar(t_menustring, '\n', t_offset, kMCStringOptionCompareExact, t_new_offset);
             }
 			
+			// if the user selects the last option, t_new_offset will not be updated, as there is no \n after the last option
```

Comment:
@livecodepanos I'm wondering if it might be more sensible to use `MCStringSplitByDelimiter` here then get the element we want.

Reply:
Thanks @montegoulding, I did not know about `MCStringSplitByDelimiter`